RecoHI/HiEgammaAlgos/plugins/AODHIPhotonProducer.cc

```diff
@@ -0,0 +1,165 @@
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/EgammaCandidates/interface/AODHIPhoton.h"
+#include "DataFormats/EgammaCandidates/interface/PhotonFwd.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+
+#include "RecoEcal/EgammaCoreTools/interface/EcalClusterLazyTools.h"
+
+#include "RecoHI/HiEgammaAlgos/interface/CxCalculator.h"
+#include "RecoHI/HiEgammaAlgos/interface/RxCalculator.h"
+#include "RecoHI/HiEgammaAlgos/interface/TxCalculator.h"
+
+#ifndef AODHIPhotonProducer_h
+#define AODHIPhotonProducer_h
+
+class AODHIPhotonProducer : public edm::stream::EDProducer<> {
+
+ public:
+
+  explicit AODHIPhotonProducer (const edm::ParameterSet& ps);
+  ~AODHIPhotonProducer();
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+ private:
+
+  virtual void beginRun (edm::Run const& r, edm::EventSetup const & es) override final;
+  virtual void endRun(edm::Run const&,  edm::EventSetup const&) override final;
+  virtual void produce(edm::Event& evt, const edm::EventSetup& es);
+
+  edm::EDGetTokenT<reco::PhotonCollection> photonProducer_;
+  edm::EDGetTokenT<EcalRecHitCollection> barrelEcalHits_;
+  edm::EDGetTokenT<EcalRecHitCollection> endcapEcalHits_;
+  edm::EDGetTokenT<HBHERecHitCollection> hbhe_;
+  edm::EDGetTokenT<HFRecHitCollection> hf_;
+  edm::EDGetTokenT<HORecHitCollection> ho_;
+  edm::EDGetTokenT<reco::BasicClusterCollection> barrelClusters_;
+  edm::EDGetTokenT<reco::BasicClusterCollection> endcapClusters_;
+  edm::EDGetTokenT<reco::TrackCollection> tracks_;
+
+  std::string trackQuality_;
+
+};
+
+#endif
+
+AODHIPhotonProducer::AODHIPhotonProducer(const edm::ParameterSet& config)
+{
+  photonProducer_   =
```

Comment:
please move all basic initializations to the initialization part (before the implementation body)